### ISSUE TITLE
KPU hangs in the middle of processing.

### ISSUE THREAD

**Issue by jkbmtx** [kendryte/nncase#92]
Hello,

recently I tried training some yolo based networks for the k210. I converted the network to the kmodel, but it hangs in between some conv operations:
```
system start
model loaded
KPUConv2D: 0.237000ms
KPUConv2D: 0.268000ms
KPUConv2D: 0.877000ms
KPUConv2D: 0.258000ms
KPUConv2D: 2.936000ms
KPUConv2D: 0.266000ms
KPUConv2D: 1.564000ms
KPUConv2D: 0.254000ms
KPUConv2D: 2.959000ms
KPUConv2D: 0.256000ms
KPUConv2D: 1.645000ms
KPUConv2D: 0.276000ms
KPUConv2D: 3.138000ms
KPUConv2D: 0.237000ms
KPUConv2D: 1.892000ms
KPUConv2D: 0.263000ms
KPUConv2D: 1.890000ms
KPUConv2D: 0.266000ms
KPUConv2D: 1.890000ms
KPUConv2D: 0.266000ms
KPUConv2D: 1.890000ms
```

the network has a couple more convs after that, I attached it. I can run the inference with ncc, I tried a couple different versions including the newest artifacts from CI. The result is always the same. The sdk should be recent, with cache fixes. I used the code that I used to run the yolo demo provided by Kendryte, I just adapted it for different resolution. Any ideas for what might be going wrong?

Thanks.
[model.zip](https://github.com/kendryte/nncase/files/4182046/model.zip)

**Comment by sunnycase** [comment]
You should use the newest SDK in develop branch.

**Comment by jkbmtx** [comment]
I switched to newest develop sdk and the issue persists, although sometimes the code proceeds without any predictions, or there is an actual crash which looks like this:

LCD init
DVP init
manuf_id:0x7fa2,device_id:0x2642
DVP interrupt config
system start
model loaded
KPUConv2D: 0.235000ms
KPUConv2D: 0.038000ms
KPUConv2D: 0.704000ms
KPUConv2D: 0.067000ms
KPUConv2D: 2.731000ms
KPUConv2D: 0.041000ms
KPUConv2D: 1.393000ms
KPUConv2D: 0.040000ms
KPUConv2D: 2.779000ms
KPUConv2D: 0.028000ms
KPUConv2D: 1.477000ms
KPUConv2D: 0.028000ms
KPUConv2D: 2.934000ms
KPUConv2D: 0.022000ms
KPUConv2D: 1.720000ms
KPUConv2D: 0.022000ms
KPUConv2D: 1.719000ms
KPUConv2D: 0.023000ms
KPUConv2D: 1.719000ms
KPUConv2D: 0.022000ms
KPUConv2D: 1.719000ms
KPUConv2D: 0.005000ms
assertion "!"Invalid memory type"" failed: file "/home/kuba/sw/ai-node-fw-stdln/sdk/kendryte-standalone-sdk/lib/nncase/runtime/interpreter.cpp", line 160, function: virtual tcb::span<unsigned char> nncase::runtime::interpreter_base::memory_at(const nncase::memory_range&) const
W (733064094) SYSCALL: sys_exit called by core 1 with 0x1

**Comment by sunnycase** [comment]
It means your model is corrupted.

**Comment by jkbmtx** [comment]
It seems like using malloc instead of static memory for model data helps. I made sure to have it with proper alignment, but probably the issue lies in the address space, ie we want the 0x4 uncached address space instead of 0x8 cached.

**Comment by jkbmtx** [comment]
Im still getting no predictions, but maybe I messed up something with the architecture, need to investigate further.

By the way, do you have any recommendations for an nn achitecture which could be useful for detecting a lot of small elements in the picture? I only need I handful of classes, one for starters, it doesnt have to be 30 fps.

**Comment by krishnak** [comment]
Do you want a ready made model or do you intend to code based on architecture or retrain?

For former there is yolov2 and mobilenet by sipeed guys

**Comment by sunnycase** [comment]
You can take a look at https://github.com/kendryte/nncase/tree/master/examples/fast_facedetect

**Comment by sunnycase** [comment]
Yolo v3 arch is supported but need compress.

**Comment by jkbmtx** [comment]
facedetect demo looks quite impressive, did you train the network? If so, would it be possible to share the framework, even under NDA? 

I need something for custom case. I generated some artificial datasets with 1-10 elements in a picture, and I used 2 open-source frameworks adapted for k210 with yolo, but the training didnt converge well at all, even though the data seemed easy enough to handle. I'm not well versed in this architecture, so it was hard for me to troubleshoot the results.
